### PR TITLE
chore: update login flow to redirect user to wahehouse selection page

### PR DESCRIPTION
**Pull request by tarashaliuk**
Updated login flow and implemented e2e tests

closes: https://spryker.atlassian.net/browse/HRZ-3116 and https://spryker.atlassian.net/browse/HRZ-3117

### CODE REVIEW

File: apps/fulfillment-e2e/src/integration/user-profile.cy.ts [hunk 1]
```diff
@@ -24,7 +22,8 @@ describe('When a user opens the user profile modal', () => {
     userProfileFragment.getWrapper().should('be.visible');
     userProfileFragment.getLogOutButton().should('be.visible');
     userProfileFragment.getNotification().should('not.exist');
-    userProfileFragment.getReceiveDataButton().should('be.visible');
+    userProfileFragment.getWrapper().should('be.visible');
```
Comment (aleespryker) [inline apps/fulfillment-e2e/src/integration/user-profile.cy.ts]: Why are these here twice?

File: libs/domain/picking/offline/data-plugin.ts [hunk 2]
```diff
@@ -45,8 +45,8 @@ export class OfflineDataPlugin implements AppPlugin {
         switchMap((authenticated) => {
           if (authenticated) {
             return this.clearDb(injector).pipe(
-              switchMap(() => this.populateDb(injector)),
-              tap(() => routerService.navigate('/'))
+              // switchMap(() => this.populateDb(injector)),
```
Comment (aleespryker) [inline libs/domain/picking/offline/data-plugin.ts]: Do we still need this?

File: libs/domain/picking/src/mixins/picking-list.mixin.ts [hunk 3]
```diff
@@ -15,9 +15,11 @@ import {
   map,
   Observable,
   switchMap,
+  tap,
```
Comment (aleespryker) [inline libs/domain/picking/src/mixins/picking-list.mixin.ts]: Do we need these imports?

File: libs/domain/picking/src/mixins/picking-list.mixin.ts [hunk 4]
```diff
@@ -15,9 +15,11 @@ import {
...
 } from 'rxjs';
 import type { PickingList } from '../models';
 import { PickingListService } from '../services';
+import { log } from '@clack/prompts';
```
Comment (aleespryker) [inline libs/domain/picking/src/mixins/picking-list.mixin.ts]: Same

File: libs/domain/picking/src/services/adapter/picking-list-default.adapter.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import { inject } from '@spryker-oryx/di';
 // Add full import because of issue with naming exports from cjs.
 import * as jsonapi from 'jsonapi-serializer';
-import { map, Observable, of, switchMap } from 'rxjs';
+import { map, Observable, of, switchMap, tap } from 'rxjs';
```
Comment (aleespryker) [inline libs/domain/picking/src/services/adapter/picking-list-default.adapter.ts]: Same

File: libs/domain/picking/picking-lists/picking-lists.component.ts [hunk 6]
```diff
@@ -10,7 +10,14 @@ import { state } from 'lit/decorators.js';
...
 import { repeat } from 'lit/directives/repeat.js';
 import { when } from 'lit/directives/when.js';
-import { distinctUntilChanged, map, startWith, Subject, switchMap } from 'rxjs';
+import {
+  distinctUntilChanged,
+  map,
+  startWith,
+  Subject,
+  switchMap,
+  tap,
```
Comment (aleespryker) [inline libs/domain/picking/picking-lists/picking-lists.component.ts]: What do we need tap for?

File: apps/fulfillment-e2e/src/integration/login.cy.ts [hunk 7]
```diff
@@ -45,14 +40,13 @@ describe('Login Suite', () => {
       oauthHandler.getTitle().should('be.visible');
       oauthHandler.getSpinner().should('be.visible');
 
-      cy.wait('@picking-lists');
+      cy.intercept('GET', '**/warehouse-user-assignments').as(
+        'warehouse-user-assignments'
+      );
+      cy.wait('@warehouse-user-assignments');
 
-      cy.location('pathname').should('be.equal', '/');
```
Comment (aleespryker) [inline apps/fulfillment-e2e/src/integration/login.cy.ts:50]: Should we check somewhere that we end up on picking lists after warehouse selection or when logging in, the warehouse is already selected?

**Comment by aleespryker** [comment]
Also check test-lint comments from github.

File: apps/fulfillment-e2e/src/integration/login.cy.ts (hunk 7 above)
Comment (tarashaliuk) [reply apps/fulfillment-e2e/src/integration/login.cy.ts:50]: The user should choose warehouse on each login. We are checking picking lists URL and picking lists visibility

File: apps/fulfillment-e2e/src/integration/user-profile.cy.ts [hunk 8]
```diff
@@ -22,7 +22,6 @@ describe('When a user opens the user profile modal', () => {
     userProfileFragment.getWrapper().should('be.visible');
     userProfileFragment.getLogOutButton().should('be.visible');
     userProfileFragment.getNotification().should('not.exist');
-    userProfileFragment.getWrapper().should('be.visible');
     userProfileFragment.getLogOutButton().should('be.visible');
```
Comment (aleespryker) [inline apps/fulfillment-e2e/src/integration/user-profile.cy.ts]: This is still duplicated

**Review by aleespryker — APPROVED** [review]
